### ISSUE TITLE
AjaxAppender does not successfully apply buffering or batching (Serilog, ASP dotnet 6)

### ISSUE THREAD

**Issue by jcreek** [mperdeck/jsnlog#196]
The example at https://jsnlog.com/Documentation/HowTo/JavascriptTryCatch does not work. Debug logs are still appearing in Serilog, despite having configured the AjaxAppender as specified.

**Comment by jcreek** [comment]
I've just confirmed that the batching doesn't appear to be working either. Included below is the code being used. Both debug messages are always sent, even when the f3 error is commented out. They are sent using separate requests and are not being batched. 

Startup.cs
```csharp
JsnlogConfiguration jsnlogConfiguration = new JsnlogConfiguration
            {
                ajaxAppenders = new List<AjaxAppender> {
                    new AjaxAppender {
                        name = "appender1",
                        storeInBufferLevel = "TRACE", // Log messages with severity smaller than TRACE are ignored
                        level = "WARN", // Log messages with severity equal or greater than TRACE and lower than WARN are stored in the internal buffer, but not sent to the server
                                      // Log messages with severity equal or greater than WARN and lower than FATAL are sent to the server on their own
                        sendWithBufferLevel = "FATAL", // Log messages with severity equal or greater than FATAL are sent to the server, along with all messages stored in the internal buffer
                        bufferSize = 20, // Stores the last up to 20 debug messages in browser memory,
                        batchSize = 20,
                        batchTimeout = 2000, // Logs are guaranteed to be sent within this period (in ms), even if the batch size has not been reached yet
                        maxBatchSize = 50 // When the server is unreachable and log messages are being stored until it is reachable again, this is the maximum number of messages that will be stored. Cannot be smaller than batchSize
                    }
                },
                loggers = new List<Logger> { // Get the loggers to use the new appender
                    new Logger {
                      appenders = "appender1"
                    }
                },
                insertJsnlogInHtmlResponses = false,
                productionLibraryPath = null
            };

            // Must be before UseStaticFiles and UseAuthorization
            app.UseJSNLog(loggerFactory, jsnlogConfiguration);
```

main.js
```js
function f1(a, b) {
    JL("f1").debug("information that may be useful when an exception happens");
}
 
function f2(c, d) {
    JL("f2").debug("more information that may be useful when an exception happens");
}
 
function f3() {
    try {
        // throw new Error("My second error");
    } catch(e) {
        JL().fatalException("exception message", e);
    }
}

f1(0,0);
f2(0,0);
f3();
```

**Comment by mperdeck** [comment]
Your server side configuration contains the line
insertJsnlogInHtmlResponses = false,

This stops JSNLog from translating your server side configuration into JavaScript config code and adding that to your pages. Without that JavaScript config code, the client side jsnlog.js library cannot pick up your server side configuration. 

So the first thing to try is to set insertJsnlogInHtmlResponses to true.

If that fails, or if you need to set insertJsnlogInHtmlResponses to false, an alternative way to generate the JavaScript config code from your server side config is to include this tag in your pages (in the HEAD, or near the top of the BODY is best):

<jl-javascript-logger-definitions />

More details on
https://jsnlog.com/Documentation/DownloadInstall

**Comment by jcreek** [comment]
Thanks for the reply :) I found that with `insertJsnlogInHtmlResponses = false,` not present the entire ASP dotnet 6 application wouldn't run at all, with some obscure exception being thrown from within JSNLog. Unfortunately I don't have the details of it to hand. I've manually added the CDN via a script tag and initialised it in my layout file as a result. 

The tag helper `<jl-javascript-logger-definitions />` just displayed as text within the project rather than rendering anything.

**Comment by mperdeck** [comment]
About getting the <jl-javascript-logger-definitions /> tag helper working:
1) If your web site does not already have a _ViewImports.cshtml file, please add an empty file with that name to your Views directory.
2) Add this line to your _ViewImports.cshtml file:
@addTagHelper "*, jsnlog"

This gets your web site to look in the jsnlog assembly for tag helpers, so it can find <jl-javascript-logger-definitions />

I reproduced the issue with setting insertJsnlogInHtmlResponses to true. Working on a fix, which may be not trivial.

**Comment by jcreek** [comment]
That has indeed got the tag helper working, which has applied the buffering and batching - thank you very much! This is a suitable workaround for the `insertJsnlogInHtmlResponses = true` issue, so I'm closing this.

**Comment by mperdeck** [comment]
FYI. The issue with insertJsnlogInHtmlResponses = true has now been fixed in the latest version (3.0.1)
